app/segments/(team)/chooseTeam.js

```diff
@@ -58,9 +63,44 @@ function ChooseTeam() {
     }
   }, [currentUserId]);
 
-  if (loading) {
-    return <Loading />;
-  }
+  useEffect(() => {
+    const unregisterAuthObserver = onIdTokenChanged(auth, async (user) => {
+      if (user) {
+        if (user.emailVerified) {
+          setVerified(true);
+          showSnackBar("Email successfully verified.");
+          clearInterval(intervalId); // Stop the interval when email is verified
+          unregisterAuthObserver(); // Unregister the auth observer
+        } else {
+          setVerified(false);
+          console.log("Error: Email Not Verified Yet, Try Again");
```

Comment:
This log along with some others spam the the console currently, unsure if we will want that when we merge this down. Might want to consolidate them a bit. 
![log spam](https://github.com/user-attachments/assets/67f190f9-f5fb-45cf-9e23-46bfcf058fa5)